From 9d6803250243c2c0c8ffa2f6ca7b36b188af219f Mon Sep 17 00:00:00 2001
From: Jan Beulich <jbeulich@suse.com>
Date: Mon, 18 Sep 2017 12:26:22 +0200
Subject: [PATCH] VMX: add new CPU families to LBR handling

Bring code up-to-date with SDM version 063, including the LBR format
enumeration.

Signed-off-by: Jan Beulich <jbeulich@suse.com>
Acked-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Kevin Tian <kevin.tian@intel.com>
diff --git a/xen/arch/x86/hvm/vmx/vmx.c b/xen/arch/x86/hvm/vmx/vmx.c
index 4d4ac48551..fd197879d5 100644
--- a/xen/arch/x86/hvm/vmx/vmx.c
+++ b/xen/arch/x86/hvm/vmx/vmx.c
@@ -2732,6 +2732,8 @@ static const struct lbr_info *last_branch_msr_get(void)
         case 0x0f:
         /* Enhanced Core */
         case 0x17:
+        /* Xeon 7400 */
+        case 0x1d:
             return c2_lbr;
         /* Nehalem */
         case 0x1a: case 0x1e: case 0x1f: case 0x2e:
@@ -2748,6 +2750,12 @@ static const struct lbr_info *last_branch_msr_get(void)
             return nh_lbr;
         /* Skylake */
         case 0x4e: case 0x5e:
+        /* Xeon Scalable */
+        case 0x55:
+        /* Cannon Lake */
+        case 0x66:
+        /* Goldmont Plus */
+        case 0x7a:
         /* Kaby Lake */
         case 0x8e: case 0x9e:
             return sk_lbr;
@@ -2757,6 +2765,8 @@ static const struct lbr_info *last_branch_msr_get(void)
         case 0x37: case 0x4a: case 0x4d: case 0x5a: case 0x5d:
         /* Xeon Phi Knights Landing */
         case 0x57:
+        /* Xeon Phi Knights Mill */
+        case 0x85:
         /* Airmont */
         case 0x4c:
             return at_lbr;
@@ -2788,6 +2798,7 @@ enum
     LBR_FORMAT_EIP_FLAGS_TSX      = 0x4, /* 64-bit EIP, Flags, TSX */
     LBR_FORMAT_EIP_FLAGS_TSX_INFO = 0x5, /* 64-bit EIP, Flags, TSX, LBR_INFO */
     LBR_FORMAT_EIP_FLAGS_CYCLES   = 0x6, /* 64-bit EIP, Flags, Cycles */
+    LBR_FORMAT_LIP_FLAGS_TSX_INFO = 0x7, /* 64-bit LIP, Flags, TSX, LBR_INFO */
 };
 
 #define LBR_FROM_SIGNEXT_2MSB  ((1ULL << 59) | (1ULL << 60))
